File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/di/module/ApplicationModule.java
package tisoul.dev.androidarchitecture.di.module;


import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import tisoul.dev.androidarchitecture.AppApplication;
import tisoul.dev.data.repository.ProfileDataRepository;
import tisoul.dev.domain.repository.ProfileRepository;

@Module
public class ApplicationModule {
    private final AppApplication application;

    public ApplicationModule(AppApplication application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Context provideApplicationContext() {
        return this.application;
    }

    @Provides
    @Singleton
    ProfileRepository provideProfileRepository(ProfileDataRepository profileRepository) {
        return profileRepository;
    }
}

File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/di/module/ApplicationModule.kt
package tisoul.dev.androidarchitecture.di.module


import android.content.Context
import dagger.Module
import dagger.Provides
import tisoul.dev.androidarchitecture.AppApplication
import tisoul.dev.data.repository.ProfileDataRepository
import tisoul.dev.domain.repository.ProfileRepository
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: AppApplication) {

    @Provides
    @Singleton
    internal fun provideApplicationContext(): Context {
        return this.application
    }

    @Provides
    @Singleton
    internal fun provideProfileRepository(profileRepository: ProfileDataRepository): ProfileRepository {
        return profileRepository
    }
}
